mwdb/resources/object.py

```diff
@@ -339,3 +339,129 @@ def get(self, type):
 
         schema = ObjectCountResponseSchema()
         return schema.dump({"count": result})
+
+
+class ObjectFavoriteResource(Resource):
+    @requires_authorization
+    def get(self, identifier):
+        """
+        ---
+        summary: Get information if object is favorite
+        description: |
+            Returns information if object is favorite
+        security:
+            - bearerAuth: []
+        tags:
+            - object
+        parameters:
+            - in: path
+              name: identifier
+              schema:
+                type: string
+              description: Object identifier
+        responses:
+            200:
+                description: List of user favorites objects
+                content:
+                  application/json:
+                    schema: ObjectFavoriteResponseSchema
+        """
+        user = g.auth_user
+        if user is None:
+            raise NotFound("Can't reference to the current user")
+
+        favorite_object = db.session.query(Object).filter(Object.dhash == identifier).first()
+        is_favorite = favorite_object.is_favorite_object(user)
+
+        schema = ObjectFavoriteResponseSchema()
+        return schema.dump({'favorite': is_favorite})
+
+    @requires_authorization
+    def post(self, identifier):
+        """
+        ---
+        summary: Add favorite object
+        description: |
+            Add an existing object as a favorite for the user.
+        security:
+            - bearerAuth: []
+        tags:
+            - object
+        parameters:
+            - in: path
+              name: identifier
+              schema:
+                type: string
+              description: Object identifier
+        responses:
+            200:
+                description: Add favorite object
+            400:
+                description: When request body is invalid
+            404:
+                description: When user or object doesn't exist.
+            409:
+                description: When object is already added as favorite.
+        """
+
+        user = g.auth_user
+        if user is None:
+            raise NotFound("Can't reference to the current user")
+
+        favorite_object = db.session.query(Object).filter(Object.dhash == identifier).first()
```

Comment:
You don't check permissions for object access. Pay attention to the security model. If you see similar flaws in other resources, check if they don't violate the permissions as well.
```suggestion
        favorite_object = Object.access(identifier)
```